### PR TITLE
Adding pipeline killing script

### PR DESCRIPTION
**Pull request by alexanderrtaylor**
Added a groovy script in order to properly stop a series of running pipelines in the proper way in order to not have pipelines in a bad state

### CODE REVIEW

File: ProperlyStopRunningPipelines.groovy [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      println("Adding: "+ job.fullName+ " build number " + build.getNumber().toInteger())
+      jobMap.put(++jobMap.size(), [ name: job.fullName, buildNum: build.getNumber().toInteger()])
+     }
+   }
+}
+use(TimeCategory)  {
+  def delay = 2.days;//Put in a Custom date here to kill anything older
+  // NO MODIFICATION
+  Jenkins.instance.getAllItems(org.jenkinsci.plugins.workflow.job.WorkflowJob).each{
+    job -> job.builds.findAll{it.isBuilding() && new Date(it.startTimeInMillis) < (new Date() - delay)}.each{
```
Comment (rsandell) [inline ProperlyStopRunningPipelines.groovy]: Will cause a lot of unnecessary disk loading: http://javadoc.jenkins.io/hudson/util/RunList.html#byTimestamp-long-long- maybe?

File: ProperlyStopRunningPipelines.groovy [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      println("Adding: "+ job.fullName+ " build number " + build.getNumber().toInteger())
+      jobMap.put(++jobMap.size(), [ name: job.fullName, buildNum: build.getNumber().toInteger()])
+     }
+   }
+ }
+println "Removing the running builds for the following jobs: "
+for(int i=1; i<= jobMap.size(); i++)
+{
+  def currentName = jobMap.get(i).name
+  def currentItem = Jenkins.instance.getAllItems(org.jenkinsci.plugins.workflow.job.WorkflowJob).find{it.name.equals(currentName)}
```
Comment (rsandell) [inline ProperlyStopRunningPipelines.groovy]: Jenkins.instance.getItemByFullName

File: ProperlyStopRunningPipelines.groovy [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import groovy.time.*;
+
+
+// A couple of example map definitions:
+
+//Used to kill all of the active jobs
+//def jobMap = [:]
+
+//Used to clean up specific builds. Put "buildNum: 0" if you want to stop the latest build
+/*def jobMap = [
```
Comment (rsandell) [inline ProperlyStopRunningPipelines.groovy]: Why use a Map when the key is just an index? Use a normal list instead and you won't have to do all the `++jobMap.size()` everywhere.

File: ProperlyStopRunningPipelines.groovy [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        Jenkins.instance.getItemByFullName(currentName).getBuildByNumber(currentBuild).doStop();
+    }
+  }
+}
+Thread.sleep(30000)
+//Calling the same as the Terminate running build command in the console log
+for(int i=1; i<= jobMap.size(); i++)
+{
+  def currentName = jobMap.get(i).name
+  def currentItem = Jenkins.instance.getAllItems(org.jenkinsci.plugins.workflow.job.WorkflowJob).find{it.name.equals(currentName)}
```
Comment (rsandell) [inline ProperlyStopRunningPipelines.groovy]: Jenkins.instance.getItemByFullName(currentName)

File: ProperlyStopRunningPipelines.groovy [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        Jenkins.instance.getItemByFullName(currentName).getBuildByNumber(currentBuild).doTerm();
+    }
+  }
+}
+Thread.sleep(30000)
+//Calling the same as the Kill running build command in the console log
+for(int i=1; i<= jobMap.size(); i++)
+{
+  def currentName = jobMap.get(i).name
+  def currentItem = Jenkins.instance.getAllItems(org.jenkinsci.plugins.workflow.job.WorkflowJob).find{it.name.equals(currentName)}
```
Comment (rsandell) [inline ProperlyStopRunningPipelines.groovy]: Jenkins.instance.getItemByFullName(currentName)

File: ProperlyStopRunningPipelines.groovy (hunk 1 above)
Comment (recampbell) [reply ProperlyStopRunningPipelines.groovy]: Right, if you force a read of every build in a system, this will take forever.

**Comment by alexanderrtaylor** [comment]
@recampbell @rsandell If I could get another quick review to see if I missed anything but I really appreciate your help

**Comment by recampbell** [comment]
LGTM, we should probably merge it

**Pull request merged**